Unity-C#/README.md

```diff
@@ -0,0 +1,256 @@
+#Preamble 
+The below style uses the MSDN Style guidelines as a base to work with Unity, however because some of the nuances in Unity this style adapts and breaks from some of the standard conventions to obtain a higher level of readability.
+
+MSDN Style: http://msdn.microsoft.com/en-us/library/ff926074.aspx
+
+MSDN Class Styles documenting that public fields are not supposed to be used: http://msdn.microsoft.com/en-us/library/ms229012(v=vs.110).aspx
+
+MSDN XML Comments: http://msdn.microsoft.com/en-us/library/b2s063f7.aspx
+
+There is also a Unity style guide however because they have used Properties, which is not supported in the Unity Editor, and the style is almost exactly the same as the MSDN style, the MSDN style was used where possible.
+Unity Style: http://wiki.unity3d.com/index.php/Csharp_Coding_Guidelines
+
+# Style
+
+## Indentation
+Yes, a highly contentious topic so let's not waste time, at Wizcorp we use **tabs** for our projects. Tabs should be used for indentation, meaning at the beginning of lines only.
+
+
+## Line length
+Limit your lines to **120 characters**. We are not living in the 90's any more our screen got a lot bigger, but at the same time our brains are the same, so let's not go too far, readability come first.
+
+
+## Whitespace
+
+* Conventional operators SHOULD be surrounded by a space (including ternary operators).
+
+* Commas SHOULD be followed by a space.
+
+* Colons SHOULD be followed by a space.
+
+* Semi-colons in for statements SHOULD be followed by a space.
+
+* Semi-colons SHOULD NOT be preceded by a space.
+
+* Function calls and method calls SHOULD NOT be followed by a space. Example: `DoSomething(string name); // NOT DoSomething (string name)`
+
+* Logical units within a block SHOULD be separated by one blank line.
+
+* Curly braces SHOULD have inner spaces.
+
+
+## Braces
+In C# where the braces are placed is up to the user, however the C# MSDN standard is on a new line.  That said as long as the project has agreed to use on the same line there should be no complaints.
+
+
+## Naming
+All names SHOULD be written in English, American English.
+### In general:
+##### Variable names
+The use of the underscore for private variables is favoured, even though it is generally not standard, is because Properties are not supported in Unity. This means that the C# standard for accessors can not be used effectively and the MSDN guidelines for not having public members is not followed.
+```csharp
+public int aName;
+private int _aName;
+```
+
+
+##### Property names
+In Unity properties are not supported in the editor, and their use is discouraged.  If you want to have a member in the Unity Editor you need to use a public member (which is against MSDN coding guidelines).
+```csharp
+public int Name { get; set; }
+```
+
+
+##### Method names
+
+```csharp
+public void Rename(int name)
+```
+
+
+##### Class names
+```csharp
+public class User
+```
+
+
+##### Constants
+```csharp
+public const int MyNumber = 42;
```

Comment:
I quite like Uppercase, I would want a difference between Const and a normal public var